tenzoMath/fanucModel.cpp

```diff
@@ -0,0 +1,326 @@
+#include "fanucModel.h"
+#include <iostream>
+#include <opencv2/core.hpp>
+#include <vector>
+#include "poly34.h"
+#define PI 3.14159265
+
+FanucModel::FanucModel()
+    :
+    RoboModel(std::vector<std::array<double, 4>>{
+        {0, 0, 150, PI / 2},
+        {0, 0, 790, 0},
+        {0, 0, 250, PI / 2},
+        {835, 0, 0, -PI / 2},
+        {0, 0, 0, PI / 2},
+        {100, 0, 0, 0},
+        {130, PI / 2, -90, 0},
+        {-190, 0, 0, 0}
+    }),
+    _toCamera((cv::Mat_<double>(4, 4) << 0, -1, 0, -43, 1, 0, 0, -90, 0, 0, 1, 130, 0, 0, 0, 1)),
+    _toSixth((cv::Mat_<double>(4, 4) << 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, -190, 0, 0, 0, 1)),
+    _forMovingToCamera((cv::Mat_<double>(4, 4) << 0, -1, 0, 43, 1, 0, 0, 90, 0, 0, 1, 130, 0, 0, 0, 1))
+{
+}
+
+std::vector<RoboModel::DhParameters> FanucModel::getDhParameters() const
+{
+    return _kinematicChain;
+}
+
+std::vector<double> FanucModel::jointsToQ(std::array<double, 6> j)
+{
+    //degrees to radians
+    for (int i = 0; i < 6; ++i)
+    {
+        j[i] *= PI / 180.0;
+    }
+    std::vector<double> q;
+    q.resize(6);
+    q[0] = j[0];
+    q[1] = -j[1] + PI / 2;
+    q[2] = j[2] + j[1];
+    q[3] = -j[3];
+    q[4] = j[4];
+    q[5] = -j[5];
+    return q;
+}
+
+cv::Mat FanucModel::fanucForwardTask(const std::array<double, 6> inputjoints)
```

Comment:
(**must be changed**): Parameter name doesn't meet our code style. See [here](https://github.com/Vasar007/FANUC/blob/master/CODESTYLE.md#naming-conventions).